core/cas-server-core-authentication/src/main/java/org/apereo/cas/authentication/handler/support/AbstractUsernamePasswordAuthenticationHandler.java

```diff
@@ -58,7 +58,6 @@ protected HandlerResult doAuthentication(final Credential credential) throws Gen
         }
 
         userPass.setUsername(transformedUsername);
-        userPass.setPassword(this.passwordEncoder.encode(userPass.getPassword()));
```

Comment:
One more thought. If password encoder is used then CAS should use **matches** method to compare it with encoded value and it does not matter which type of of encoder is used NONE|DEFAULT|STANDARD|BCRYPT . so fix should ensure **matches** are used always and encoding of password in abstract class should be removed (my opinion). Only cases where there is no passwordEncoder used like for example in **QueryAndEncodeDatabaseAuthenticationHandler** and there is other routine like _digestEncodedPassword_ should CAS use it instead of passwordEncoder. My personal opinion they should be changed to implement PasswordEncoder as well to unify security and way how to compare password.

 What is your opinion?